### PR TITLE
Remove unused TracingPlugin interface

### PR DESCRIPTION
**Pull request by DaveCTurner**

### CODE REVIEW

**Comment by DaveCTurner** [comment]
Turns out we're not really using this, so let's drop it.

**Review by AthenaEryma — APPROVED** [review]
LGTM!

File: server/src/main/java/org/elasticsearch/tasks/Traceable.java [hunk 1]
```diff
@@ -6,23 +6,14 @@
  * Side Public License, v 1.
  */
 
-package org.elasticsearch.plugins;
+package org.elasticsearch.tasks;
```
Comment (idegtiarenko) [inline server/src/main/java/org/elasticsearch/tasks/Traceable.java:9]: I wonder if we want to make a dedicated package for tracing (as cluster state and auth tracing is not directly related to tasks)?

Comment (DaveCTurner) [reply server/src/main/java/org/elasticsearch/tasks/Traceable.java]: ++ moved to `org.elasticsearch.tracing` in f73b052ff25963590c329f719d778333f2eee503.

**Comment by tlrx** [comment]
I come late to the party but LGTM - This is a good change

**Pull request merged**